addons/binding/org.openhab.binding.samsungtv/src/main/java/org/openhab/binding/samsungtv/internal/protocol/WebSocketRemote.java

```diff
@@ -0,0 +1,208 @@
+            App[] data;
+
+            String id;
+            String token;
+        };
+
+        Data data;
+
+        static class Params {
+            String params;
+
+            static class Data {
+                String appId;
+            };
+
+            Data data;
+        };
+
+        Params params;
+
+    }
+
+    /**
+     * @param remoteControllerWebSocket
+     */
+    WebSocketRemote(RemoteControllerWebSocket remoteControllerWebSocket) {
+        super(remoteControllerWebSocket);
+    }
+
+    @Override
+    public void onWebSocketError(Throwable error) {
+        super.onWebSocketError(error);
+        remoteControllerWebSocket.callback.connectionError(error);
+    }
+
+    @Override
+    public void onWebSocketText(String msgarg) {
+        String msg = msgarg.replace('\n', ' ');
+        super.onWebSocketText(msg);
+        try {
+            JSONMessage jsonMsg = remoteControllerWebSocket.gson.fromJson(msg, JSONMessage.class);
+            switch (jsonMsg.event) {
+                case "ms.channel.connect":
+                    logger.debug("Remote channel connected. Token = {}", jsonMsg.data.token);
+                    if (jsonMsg.data.token != null) {
+                        this.remoteControllerWebSocket.callback.putConfig(SamsungTvConfiguration.WEBSOCKET_TOKEN,
+                                jsonMsg.data.token);
+                    }
+                    getApps();
+                    break;
+                case "ms.channel.clientConnect":
+                    logger.debug("Remote client connected");
+                    break;
+                case "ms.channel.clientDisconnect":
+                    logger.debug("Remote client disconnected");
+                    break;
+                case "ed.edenTV.update":
+                    logger.debug("edenTV update: {}", jsonMsg.data.update_type);
+                    remoteControllerWebSocket.updateCurrentApp();
+                    break;
+                case "ed.apps.launch":
+                    logger.debug("App launched: {}", jsonMsg.params.data.appId);
+                    break;
+                case "ed.installedApp.get":
+                    handleInstalledApps(jsonMsg);
+                    break;
+                default:
+                    logger.debug("WebSocketRemote Unknown event: {}", msg);
+
+            }
+        } catch (Exception e) {
+            logger.warn("{}: Error ({}) in message: {}", this.getClass().getSimpleName(), e.getMessage(), msg, e);
+        }
+    }
+
+    private void handleInstalledApps(JSONMessage jsonMsg) {
+        remoteControllerWebSocket.apps.clear();
+
+        for (JSONMessage.App jsonApp : jsonMsg.data.data) {
+            App app = remoteControllerWebSocket.new App(jsonApp.appId, jsonApp.name, jsonApp.app_type);
+            remoteControllerWebSocket.apps.put(app.name, app);
+        }
+
+        logger.debug("Installed Apps: " + remoteControllerWebSocket.apps.entrySet().stream()
```

Comment:
Since this is not simple logging and many string operations it makes sense to put this inside an
 `if (logger.isDebugEnabled()) { .. }` block.